### PR TITLE
kvserver/rangefeed: return Disconnector to node.MuxRangefeed

### PR DESCRIPTION
**Pull request by wenyihu6**
This patch updates p.Register to return each registration as a `Disconnector` up
the stack to `node.MuxRangefeed`. While it's currently unused, future commits
will use the `Disconnector` to disconnect rangefeeds at the node level,
replacing the current approach of canceling each stream’s context to disconnect
each rangefeed.

Part of: https://github.com/cockroachdb/cockroach/issues/110432
Release note: none

Co-authored-by: Steven Danna danna@cockroachlabs.com

### CODE REVIEW

**Comment by wenyihu6** [comment]
Feel free to hold off on reviews until the previous ones are merged if you prefer.

**Review by tbg — APPROVED** [review]

**Comment by wenyihu6** [comment]
TFTR!

bors r=tbg

**Pull request merged**
